### ISSUE TITLE
Update `requests` library to >= 2.31.0

### ISSUE THREAD

**Issue by Selutario** [wazuh/wazuh#17283]
## Description

Due to a vulnerability detected in `requests` library prior to `2.31.0`, it needs to be updated at least to version **2.31.0** or later, while checking its compatibility with other libraries installed in Wazuh. Requests is a requirement of libraries used in GCP, Azure and DockerListener wodles.

This is the dependencies tree for 4.4:
```
aiohttp-cache==2.2.0
  - aiohttp [required: >=3.6,<4.0, installed: 3.8.1]
    - aiosignal [required: >=1.1.2, installed: 1.2.0]
      - frozenlist [required: >=1.1.0, installed: 1.2.0]
    - async-timeout [required: >=4.0.0a3,<5.0, installed: 4.0.2]
    - attrs [required: >=17.3.0, installed: 20.3.0]
    - charset-normalizer [required: >=2.0,<3.0, installed: 2.0.4]
    - frozenlist [required: >=1.1.1, installed: 1.2.0]
    - multidict [required: >=4.5,<7.0, installed: 5.1.0]
    - yarl [required: >=1.0,<2.0, installed: 1.6.3]
      - idna [required: >=2.0, installed: 2.9]
      - multidict [required: >=4.0, installed: 5.1.0]
  - aioredis [required: >=1.3,<2.0, installed: 1.3.1]
    - async-timeout [required: Any, installed: 4.0.2]
    - hiredis [required: Any, installed: 1.1.0]
  - envparse [required: >=0.2.0,<0.3.0, installed: 0.2.0]
aiohttp-cors==0.7.0
  - aiohttp [required: >=1.1, installed: 3.8.1]
    - aiosignal [required: >=1.1.2, installed: 1.2.0]
      - frozenlist [required: >=1.1.0, installed: 1.2.0]
    - async-timeout [required: >=4.0.0a3,<5.0, installed: 4.0.2]
    - attrs [required: >=17.3.0, installed: 20.3.0]
    - charset-normalizer [required: >=2.0,<3.0, installed: 2.0.4]
    - frozenlist [required: >=1.1.1, installed: 1.2.0]
    - multidict [required: >=4.5,<7.0, installed: 5.1.0]
    - yarl [required: >=1.0,<2.0, installed: 1.6.3]
      - idna [required: >=2.0, installed: 2.9]
      - multidict [required: >=4.0, installed: 5.1.0]
aiohttp-jinja2==1.4.2
  - aiohttp [required: >=3.6.3, installed: 3.8.1]
    - aiosignal [required: >=1.1.2, installed: 1.2.0]
      - frozenlist [required: >=1.1.0, installed: 1.2.0]
    - async-timeout [required: >=4.0.0a3,<5.0, installed: 4.0.2]
    - attrs [required: >=17.3.0, installed: 20.3.0]
    - charset-normalizer [required: >=2.0,<3.0, installed: 2.0.4]
    - frozenlist [required: >=1.1.1, installed: 1.2.0]
    - multidict [required: >=4.5,<7.0, installed: 5.1.0]
    - yarl [required: >=1.0,<2.0, installed: 1.6.3]
      - idna [required: >=2.0, installed: 2.9]
      - multidict [required: >=4.0, installed: 5.1.0]
  - jinja2 [required: >=2.10.1, installed: 3.0.0]
    - MarkupSafe [required: >=2.0.0rc2, installed: 2.1.2]
  - typing-extensions [required: >=3.7.4, installed: 3.10.0.2]
api==4.6.0
asn1crypto==1.3.0
azure-storage-blob==2.1.0
  - azure-common [required: >=1.1.5, installed: 1.1.25]
  - azure-storage-common [required: ~=2.1, installed: 2.1.0]
    - azure-common [required: >=1.1.5, installed: 1.1.25]
    - cryptography [required: Any, installed: 3.3.2]
      - cffi [required: >=1.12, installed: 1.14.4]
        - pycparser [required: Any, installed: 2.20]
      - six [required: >=1.4.1, installed: 1.14.0]
    - python-dateutil [required: Any, installed: 2.8.1]
      - six [required: >=1.5, installed: 1.14.0]
    - requests [required: Any, installed: 2.25.1]
      - certifi [required: >=2017.4.17, installed: 2022.12.7]
      - chardet [required: >=3.0.2,<5, installed: 3.0.4]
      - idna [required: >=2.5,<3, installed: 2.9]
      - urllib3 [required: >=1.21.1,<1.27, installed: 1.26.5]
boto3==1.17.85
  - botocore [required: >=1.20.85,<1.21.0, installed: 1.20.85]
    - jmespath [required: >=0.7.1,<1.0.0, installed: 0.9.5]
    - python-dateutil [required: >=2.1,<3.0.0, installed: 2.8.1]
      - six [required: >=1.5, installed: 1.14.0]
    - urllib3 [required: >=1.25.4,<1.27, installed: 1.26.5]
  - jmespath [required: >=0.7.1,<1.0.0, installed: 0.9.5]
  - s3transfer [required: >=0.4.0,<0.5.0, installed: 0.4.2]
    - botocore [required: >=1.12.36,<2.0a.0, installed: 1.20.85]
      - jmespath [required: >=0.7.1,<1.0.0, installed: 0.9.5]
      - python-dateutil [required: >=2.1,<3.0.0, installed: 2.8.1]
        - six [required: >=1.5, installed: 1.14.0]
      - urllib3 [required: >=1.25.4,<1.27, installed: 1.26.5]
connexion==2.6.0
  - clickclick [required: >=1.2, installed: 20.10.2]
    - click [required: >=4.0, installed: 8.1.3]
    - PyYAML [required: >=3.11, installed: 5.4.1]
  - flask [required: >=1.0.4, installed: 2.2.5]
    - click [required: >=8.0, installed: 8.1.3]
    - importlib-metadata [required: >=3.6.0, installed: 3.10.1]
      - zipp [required: >=0.5, installed: 3.3.2]
    - itsdangerous [required: >=2.0, installed: 2.0.0]
    - Jinja2 [required: >=3.0, installed: 3.0.0]
      - MarkupSafe [required: >=2.0.0rc2, installed: 2.1.2]
    - Werkzeug [required: >=2.2.2, installed: 2.2.3]
      - MarkupSafe [required: >=2.1.1, installed: 2.1.2]
  - inflection [required: >=0.3.1, installed: 0.3.1]
  - jsonschema [required: >=2.5.1, installed: 2.6.0]
  - openapi-spec-validator [required: >=0.2.4, installed: 0.2.6]
    - jsonschema [required: <3, installed: 2.6.0]
    - pathlib [required: Any, installed: 1.0.1]
    - PyYAML [required: >=3.13, installed: 5.4.1]
    - six [required: Any, installed: 1.14.0]
  - PyYAML [required: >=5.1, installed: 5.4.1]
  - requests [required: >=2.9.1, installed: 2.25.1]
    - certifi [required: >=2017.4.17, installed: 2022.12.7]
    - chardet [required: >=3.0.2,<5, installed: 3.0.4]
    - idna [required: >=2.5,<3, installed: 2.9]
    - urllib3 [required: >=1.21.1,<1.27, installed: 1.26.5]
Cython==0.29.21
defusedxml==0.6.0
docker==4.2.0
  - requests [required: >=2.14.2,!=2.18.0, installed: 2.25.1]
    - certifi [required: >=2017.4.17, installed: 2022.12.7]
    - chardet [required: >=3.0.2,<5, installed: 3.0.4]
    - idna [required: >=2.5,<3, installed: 2.9]
    - urllib3 [required: >=1.21.1,<1.27, installed: 1.26.5]
  - six [required: >=1.4.0, installed: 1.14.0]
  - websocket-client [required: >=0.32.0, installed: 0.57.0]
    - six [required: Any, installed: 1.14.0]
docker-pycreds==0.4.0
  - six [required: >=1.4.0, installed: 1.14.0]
docutils==0.15.2
freezegun==1.2.2
  - python-dateutil [required: >=2.7, installed: 2.8.1]
    - six [required: >=1.5, installed: 1.14.0]
future==0.18.3
google-cloud-pubsub==2.7.1
  - google-api-core [required: >=1.26.0,<3.0.0dev, installed: 1.30.0]
    - google-auth [required: >=1.25.0,<2.0dev, installed: 1.28.0]
      - cachetools [required: >=2.0.0,<5.0, installed: 4.1.0]
      - pyasn1-modules [required: >=0.2.1, installed: 0.2.8]
        - pyasn1 [required: >=0.4.6,<0.5.0, installed: 0.4.8]
      - rsa [required: >=3.1.4,<5, installed: 4.7.2]
        - pyasn1 [required: >=0.1.3, installed: 0.4.8]
      - setuptools [required: >=40.3.0, installed: 65.5.1]
      - six [required: >=1.9.0, installed: 1.14.0]
    - googleapis-common-protos [required: >=1.6.0,<2.0dev, installed: 1.51.0]
      - protobuf [required: >=3.6.0, installed: 3.19.6]
    - packaging [required: >=14.3, installed: 20.9]
      - pyparsing [required: >=2.0.2, installed: 2.4.7]
    - protobuf [required: >=3.12.0, installed: 3.19.6]
    - pytz [required: Any, installed: 2020.1]
    - requests [required: >=2.18.0,<3.0.0dev, installed: 2.25.1]
      - certifi [required: >=2017.4.17, installed: 2022.12.7]
      - chardet [required: >=3.0.2,<5, installed: 3.0.4]
      - idna [required: >=2.5,<3, installed: 2.9]
      - urllib3 [required: >=1.21.1,<1.27, installed: 1.26.5]
    - setuptools [required: >=40.3.0, installed: 65.5.1]
    - six [required: >=1.13.0, installed: 1.14.0]
  - grpc-google-iam-v1 [required: >=0.12.3,<0.13dev, installed: 0.12.3]
    - googleapis-common-protos [required: >=1.5.2,<2.0.0dev, installed: 1.51.0]
      - protobuf [required: >=3.6.0, installed: 3.19.6]
    - grpcio [required: >=1.0.0,<2.0.0dev, installed: 1.38.1]
      - six [required: >=1.5.2, installed: 1.14.0]
  - grpcio [required: >=1.38.1,<2.0dev, installed: 1.38.1]
    - six [required: >=1.5.2, installed: 1.14.0]
  - libcst [required: >=0.3.10, installed: 0.3.20]
    - pyyaml [required: >=5.2, installed: 5.4.1]
    - typing-extensions [required: >=3.7.4.2, installed: 3.10.0.2]
    - typing-inspect [required: >=0.4.0, installed: 0.7.1]
      - mypy-extensions [required: >=0.3.0, installed: 0.4.3]
      - typing-extensions [required: >=3.7.4, installed: 3.10.0.2]
  - packaging [required: >=14.3, installed: 20.9]
    - pyparsing [required: >=2.0.2, installed: 2.4.7]
  - proto-plus [required: >=1.7.1, installed: 1.19.0]
    - protobuf [required: >=3.12.0, installed: 3.19.6]
google-cloud-storage==1.39.0
  - google-auth [required: >=1.11.0,<2.0dev, installed: 1.28.0]
    - cachetools [required: >=2.0.0,<5.0, installed: 4.1.0]
    - pyasn1-modules [required: >=0.2.1, installed: 0.2.8]
      - pyasn1 [required: >=0.4.6,<0.5.0, installed: 0.4.8]
    - rsa [required: >=3.1.4,<5, installed: 4.7.2]
      - pyasn1 [required: >=0.1.3, installed: 0.4.8]
    - setuptools [required: >=40.3.0, installed: 65.5.1]
    - six [required: >=1.9.0, installed: 1.14.0]
  - google-cloud-core [required: >=1.4.1,<2.0dev, installed: 1.7.1]
    - google-api-core [required: >=1.21.0,<2.0.0dev, installed: 1.30.0]
      - google-auth [required: >=1.25.0,<2.0dev, installed: 1.28.0]
        - cachetools [required: >=2.0.0,<5.0, installed: 4.1.0]
        - pyasn1-modules [required: >=0.2.1, installed: 0.2.8]
          - pyasn1 [required: >=0.4.6,<0.5.0, installed: 0.4.8]
        - rsa [required: >=3.1.4,<5, installed: 4.7.2]
          - pyasn1 [required: >=0.1.3, installed: 0.4.8]
        - setuptools [required: >=40.3.0, installed: 65.5.1]
        - six [required: >=1.9.0, installed: 1.14.0]
      - googleapis-common-protos [required: >=1.6.0,<2.0dev, installed: 1.51.0]
        - protobuf [required: >=3.6.0, installed: 3.19.6]
      - packaging [required: >=14.3, installed: 20.9]
        - pyparsing [required: >=2.0.2, installed: 2.4.7]
      - protobuf [required: >=3.12.0, installed: 3.19.6]
      - pytz [required: Any, installed: 2020.1]
      - requests [required: >=2.18.0,<3.0.0dev, installed: 2.25.1]
        - certifi [required: >=2017.4.17, installed: 2022.12.7]
        - chardet [required: >=3.0.2,<5, installed: 3.0.4]
        - idna [required: >=2.5,<3, installed: 2.9]
        - urllib3 [required: >=1.21.1,<1.27, installed: 1.26.5]
      - setuptools [required: >=40.3.0, installed: 65.5.1]
      - six [required: >=1.13.0, installed: 1.14.0]
    - google-auth [required: >=1.24.0,<2.0dev, installed: 1.28.0]
      - cachetools [required: >=2.0.0,<5.0, installed: 4.1.0]
      - pyasn1-modules [required: >=0.2.1, installed: 0.2.8]
        - pyasn1 [required: >=0.4.6,<0.5.0, installed: 0.4.8]
      - rsa [required: >=3.1.4,<5, installed: 4.7.2]
        - pyasn1 [required: >=0.1.3, installed: 0.4.8]
      - setuptools [required: >=40.3.0, installed: 65.5.1]
      - six [required: >=1.9.0, installed: 1.14.0]
    - six [required: >=1.12.0, installed: 1.14.0]
  - google-resumable-media [required: >=1.3.0,<2.0dev, installed: 1.3.1]
    - google-crc32c [required: >=1.0,<2.0dev, installed: 1.1.2]
    - six [required: >=1.4.0, installed: 1.14.0]
  - requests [required: >=2.18.0,<3.0.0dev, installed: 2.25.1]
    - certifi [required: >=2017.4.17, installed: 2022.12.7]
    - chardet [required: >=3.0.2,<5, installed: 3.0.4]
    - idna [required: >=2.5,<3, installed: 2.9]
    - urllib3 [required: >=1.21.1,<1.27, installed: 1.26.5]
more-itertools==8.2.0
pip==23.0.1
pipdeptree==2.7.1
psutil==5.8.0
pyarrow==8.0.0
  - numpy [required: >=1.16.6, installed: 1.21.6]
pydevd-pycharm==232.5150.120
python-jose==3.1.0
  - ecdsa [required: <1.0, installed: 0.16.1]
    - six [required: >=1.9.0, installed: 1.14.0]
  - pyasn1 [required: Any, installed: 0.4.8]
  - rsa [required: Any, installed: 4.7.2]
    - pyasn1 [required: >=0.1.3, installed: 0.4.8]
  - six [required: <2.0, installed: 1.14.0]
python-json-logger==2.0.2
secure==0.2.1
SQLAlchemy==1.3.11
tabulate==0.8.9
uvloop==0.15.2
wazuh==4.6.0
xmltodict==0.12.0
```

**Comment by EduLeon12** [comment]
## Issue Update

First we tested the UTs behavior after upgrading the request dependency to version `2.31.0`

<details> <summary> Unit Test </summary>

- API:

```
(request-upgrade) eduardoleon@pop-os:~/git/wazuh$ pytest api/api/ --disable-warnings
====================================== test session starts ======================================
platform linux -- Python 3.9.16, pytest-7.3.1, pluggy-1.2.0
rootdir: /home/eduardoleon/git/wazuh/api
plugins: html-2.1.1, tavern-1.23.5, aiohttp-1.0.4, trio-0.7.0, metadata-3.0.0, asyncio-0.18.1
asyncio: mode=auto
collected 560 items

api/api/controllers/test/test_active_response_controller.py .                             [  0%]
api/api/controllers/test/test_agent_controller.py ....................................... [  7%]
.....                                                                                     [  8%]
api/api/controllers/test/test_cdb_list_controller.py ......                               [  9%]
api/api/controllers/test/test_ciscat_controller.py .                                      [  9%]
api/api/controllers/test/test_cluster_controller.py ........................              [ 13%]
api/api/controllers/test/test_decoder_controller.py .......                               [ 14%]
api/api/controllers/test/test_default_controller.py .                                     [ 15%]
api/api/controllers/test/test_event_controller.py .                                       [ 15%]
api/api/controllers/test/test_experimental_controller.py ...............                  [ 17%]
api/api/controllers/test/test_manager_controller.py ..................                    [ 21%]
api/api/controllers/test/test_mitre_controller.py .......                                 [ 22%]
api/api/controllers/test/test_overview_controller.py .                                    [ 22%]
api/api/controllers/test/test_rootcheck_controller.py ....                                [ 23%]
api/api/controllers/test/test_rule_controller.py ........                                 [ 24%]
api/api/controllers/test/test_sca_controller.py ..                                        [ 25%]
api/api/controllers/test/test_security_controller.py .................................... [ 31%]
...............                                                                           [ 34%]
api/api/controllers/test/test_syscheck_controller.py ....                                 [ 34%]
api/api/controllers/test/test_syscollector_controller.py .........                        [ 36%]
api/api/controllers/test/test_task_controller.py .                                        [ 36%]
api/api/controllers/test/test_vulnerability_controller.py ....                            [ 37%]
api/api/models/test/test_model.py ..............................                          [ 42%]
api/api/test/test_alogging.py ......................                                      [ 46%]
api/api/test/test_authentication.py ...........                                           [ 48%]
api/api/test/test_configuration.py .............................................          [ 56%]
api/api/test/test_encoder.py ...                                                          [ 57%]
api/api/test/test_middlewares.py ...............                                          [ 59%]
api/api/test/test_uri_parser.py ...                                                       [ 60%]
api/api/test/test_util.py ..............................................                  [ 68%]
api/api/test/test_validator.py .......................................................... [ 78%]
......................................................................................... [ 94%]
.............................                                                             [100%]

=============================== 560 passed, 17 warnings in 1.70s ================================
```

- Framework:

```
(request-upgrade) eduardoleon@pop-os:~/git/wazuh$ pytest framework/ --disable-warnings
====================================== test session starts ======================================
platform linux -- Python 3.9.16, pytest-7.3.1, pluggy-1.2.0
rootdir: /home/eduardoleon/git/wazuh/framework
plugins: html-2.1.1, tavern-1.23.5, aiohttp-1.0.4, trio-0.7.0, metadata-3.0.0, asyncio-0.18.1
asyncio: mode=auto
collected 2174 items

framework/scripts/tests/test_agent_groups.py ..............                               [  0%]
framework/scripts/tests/test_agent_upgrade.py ...............                             [  1%]
framework/scripts/tests/test_cluster_control.py ......                                    [  1%]
framework/scripts/tests/test_rbac_control.py .........                                    [  2%]
framework/scripts/tests/test_wazuh_clusterd.py .......                                    [  2%]
framework/scripts/tests/test_wazuh_logtest.py ......................                      [  3%]
framework/wazuh/core/cluster/dapi/tests/test_dapi.py ................................     [  4%]
framework/wazuh/core/cluster/tests/test_client.py ................                        [  5%]
framework/wazuh/core/cluster/tests/test_cluster.py ...................................    [  7%]
framework/wazuh/core/cluster/tests/test_common.py ....................................... [  8%]
...........................................                                               [ 10%]
framework/wazuh/core/cluster/tests/test_control.py ......                                 [ 11%]
framework/wazuh/core/cluster/tests/test_local_client.py ..............                    [ 11%]
framework/wazuh/core/cluster/tests/test_local_server.py ........................          [ 12%]
framework/wazuh/core/cluster/tests/test_master.py ....................................... [ 14%]
........                                                                                  [ 15%]
framework/wazuh/core/cluster/tests/test_server.py .............................           [ 16%]
framework/wazuh/core/cluster/tests/test_utils.py ...........                              [ 16%]
framework/wazuh/core/cluster/tests/test_worker.py ..................................      [ 18%]
framework/wazuh/core/tests/test_active_response.py ..............                         [ 19%]
framework/wazuh/core/tests/test_agent.py ................................................ [ 21%]
......................................................................................... [ 25%]
............                                                                              [ 26%]
framework/wazuh/core/tests/test_cdb_list.py ......................................        [ 27%]
framework/wazuh/core/tests/test_common.py .........                                       [ 28%]
framework/wazuh/core/tests/test_configuration.py ........................................ [ 30%]
...............................                                                           [ 31%]
framework/wazuh/core/tests/test_database.py .............                                 [ 32%]
framework/wazuh/core/tests/test_decoder.py ................                               [ 32%]
framework/wazuh/core/tests/test_exception.py ..........                                   [ 33%]
framework/wazuh/core/tests/test_input_validator.py ...                                    [ 33%]
framework/wazuh/core/tests/test_logtest.py ..                                             [ 33%]
framework/wazuh/core/tests/test_manager.py ................                               [ 34%]
framework/wazuh/core/tests/test_mitre.py .............                                    [ 34%]
framework/wazuh/core/tests/test_pyDaemonModule.py .....                                   [ 35%]
framework/wazuh/core/tests/test_results.py ........................................       [ 36%]
framework/wazuh/core/tests/test_rootcheck.py .............                                [ 37%]
framework/wazuh/core/tests/test_rule.py ......................                            [ 38%]
framework/wazuh/core/tests/test_sca.py ........................                           [ 39%]
framework/wazuh/core/tests/test_security.py .............                                 [ 40%]
framework/wazuh/core/tests/test_stats.py .................                                [ 40%]
framework/wazuh/core/tests/test_syscheck.py .......                                       [ 41%]
framework/wazuh/core/tests/test_syscollector.py ...                                       [ 41%]
framework/wazuh/core/tests/test_task.py ........                                          [ 41%]
framework/wazuh/core/tests/test_utils.py ................................................ [ 44%]
......................................................................................... [ 48%]
......................................................................................... [ 52%]
...................................                                                       [ 53%]
framework/wazuh/core/tests/test_vulnerability.py ..                                       [ 53%]
framework/wazuh/core/tests/test_wazuh_queue.py .......................                    [ 54%]
framework/wazuh/core/tests/test_wazuh_socket.py ....................                      [ 55%]
framework/wazuh/core/tests/test_wdb.py ...............................                    [ 57%]
framework/wazuh/core/tests/test_wlogging.py ............                                  [ 57%]
framework/wazuh/rbac/tests/test_auth_context.py ..                                        [ 57%]
framework/wazuh/rbac/tests/test_decorators.py ........................................... [ 59%]
..................................................................                        [ 62%]
framework/wazuh/rbac/tests/test_default_configuration.py ................................ [ 64%]
.........................                                                                 [ 65%]
framework/wazuh/rbac/tests/test_orm.py .................................................. [ 67%]
...............                                                                           [ 68%]
framework/wazuh/rbac/tests/test_preprocessor.py ...........                               [ 69%]
framework/wazuh/tests/test_active_response.py ............                                [ 69%]
framework/wazuh/tests/test_agent.py ..................................................... [ 72%]
...................................................................                       [ 75%]
framework/wazuh/tests/test_cdb_list.py .................................................. [ 77%]
...                                                                                       [ 77%]
framework/wazuh/tests/test_ciscat.py .................................                    [ 79%]
framework/wazuh/tests/test_cluster.py ..........                                          [ 79%]
framework/wazuh/tests/test_decoder.py ................................................... [ 81%]
.......                                                                                   [ 82%]
framework/wazuh/tests/test_event.py ....                                                  [ 82%]
framework/wazuh/tests/test_group.py .......                                               [ 82%]
framework/wazuh/tests/test_logtest.py ......                                              [ 83%]
framework/wazuh/tests/test_manager.py ....................................                [ 84%]
framework/wazuh/tests/test_mitre.py .......                                               [ 85%]
framework/wazuh/tests/test_rootcheck.py ................................................. [ 87%]
.                                                                                         [ 87%]
framework/wazuh/tests/test_rule.py ...................................................... [ 89%]
....................                                                                      [ 90%]
framework/wazuh/tests/test_sca.py ...........                                             [ 91%]
framework/wazuh/tests/test_security.py .................................................. [ 93%]
.....................                                                                     [ 94%]
framework/wazuh/tests/test_stats.py ...............                                       [ 95%]
framework/wazuh/tests/test_syscheck.py .........................                          [ 96%]
framework/wazuh/tests/test_syscollector.py ............                                   [ 96%]
framework/wazuh/tests/test_task.py ............................                           [ 98%]
framework/wazuh/tests/test_vulnerability.py ........................................      [100%]

========================= 2174 passed, 52 warnings in 189.53s (0:03:09) =========================
```


 </details>

After investigating the process to update the dependency, The pre-installed cpython with the upgraded request library has been created. 
Testing on a ubuntu 22.04 container had a successful result but since the earliest supported distro is centOS 7, this will be tested on amd and aarch centOS 7 images and results published to guarantee compatibility.

**Comment by EduLeon12** [comment]
## Issue Update

The precompiladed-cpython for the amd x86 architecture has been successfully created and tested in a local environment.

using that we will be able to use the necessary files to create the same precompiled version for the aarch architectura nd the source cpython.

Once that is done and tested the process will be thoroughly explained here.

**Comment by EduLeon12** [comment]
## Issue Update

Source and pre-compiled cpython has been generated successfully and queued for testing of the pre-install generation.

### Walkthrough

To generate the cpythons the process was the following:

To test a centOS 7 version was used for both architectures (AMD, AARCH) since this is the earliest that could cause a compatibility issue.

After using a pre-built centOS 7 image with all necessary packages to build Wazuh from source.
a cpython file was created executing the `make deps` command in `/wazuh/src/`. The cpython generated was decompressed, changed and mapped using [pip2pi](https://github.com/BraulioV/pip2pi) and the [new package version wheel ](https://pypi.org/project/requests/#files). 

After changing the `wazuh/framework/requirements.txt` file to target version `request=2.31.0`  abd proceeding with the installation of wazuh, we test the installation to ensure everything is working properly, if no error is shown then the generated cpython file is the pre-compiled_cpython for that arquitecture and it's the one uploaded for each arquitecture.

Below is the output of the installation of wazuh from source using the pre-compiled cpython for each arquitecture.

<details> <summary> Amd </summary>

```
[root@4c1b0c4828be /]# uname -a
Linux 4c1b0c4828be 6.4.6-76060406-generic #202307241739~1690928105~22.04~d567a38 SMP PREEMPT_DYNAMIC Tue A x86_64 x86_64 x86_64 GNU/Linux
[root@4c1b0c4828be /]# /var/ossec/bin/wazuh-control status
wazuh-clusterd not running...
wazuh-modulesd is running...
wazuh-monitord is running...
wazuh-logcollector is running...
wazuh-remoted is running...
wazuh-syscheckd is running...
wazuh-analysisd is running...
wazuh-maild not running...
wazuh-execd is running...
wazuh-db is running...
wazuh-authd is running...
wazuh-agentlessd not running...
wazuh-integratord not running...
wazuh-dbd not running...
wazuh-csyslogd not running...
wazuh-apid is running...

```

</details>

<details> <summary> Aarch </summary>

```
[root@178ffd9d877b /]# uname -a
Linux 178ffd9d877b 6.4.6-76060406-generic #202307241739~1690928105~22.04~d567a38 SMP PREEMPT_DYNAMIC Tue A aarch64 aarch64 aarch64 GNU/Linux

[root@178ffd9d877b /]# /var/ossec/bin/wazuh-control status
wazuh-clusterd not running...
wazuh-modulesd is running...
wazuh-monitord is running...
wazuh-logcollector is running...
wazuh-remoted is running...
wazuh-syscheckd is running...
wazuh-analysisd is running...
wazuh-maild not running...
wazuh-execd is running...
wazuh-db is running...
wazuh-authd is running...
wazuh-agentlessd not running...
wazuh-integratord not running...
wazuh-dbd not running...
wazuh-csyslogd not running...
wazuh-apid is running...
```

</details>

**Comment by EduLeon12** [comment]
## Issue Update

Issue will be Blocked until https://github.com/wazuh/wazuh/issues/13265 is resumed, After that it will be decided if a PR will be done to Epic branch or the upgrade will be done directly on epic branch.

**Comment by EduLeon12** [comment]
## Issue Update

Will be merged as a security patch with #18214.

All testing was re done and can be seen [here](https://github.com/wazuh/wazuh/issues/18214#issuecomment-1776079506).
